modules/fedimint-ln-common/src/lib.rs

```diff
@@ -245,6 +252,31 @@ pub struct LightningGatewayRegistration {
     pub valid_until: SystemTime,
 }
 
+impl Encodable for LightningGatewayRegistration {
+    fn consensus_encode<W: Write>(&self, writer: &mut W) -> Result<usize, Error> {
+        let json_repr = serde_json::to_string(self).map_err(|e| {
+            Error::new(
+                ErrorKind::Other,
+                format!("Failed to serialize LightningGatewayRegistration: {}", e),
+            )
+        })?;
+
+        json_repr.consensus_encode(writer)
+    }
+}
+
+impl Decodable for LightningGatewayRegistration {
+    fn consensus_decode<R: Read>(
+        r: &mut R,
+        modules: &ModuleDecoderRegistry,
+    ) -> Result<Self, DecodeError> {
+        let json_repr = String::consensus_decode(r, modules)?;
+        serde_json::from_str(&json_repr).map_err(|_| {
+            DecodeError::from_str("Failed to deserialize LightningGatewayRegistration")
```

Comment:
Include the error?

Reply:
Neither `from_str` or `from_err` would work if I wanted to add context, but I just found out about `new_custom`! That way it's possible.